### ISSUE TITLE
Adding ADF to the Database?

### ISSUE THREAD

**Issue by ransom1122** [midwan/amiberry#106]
Hi,

I was wondering if ADF games can be added to the database of games along with the WHDLoad & CD32?

It would be great if we can add the odd game or two that that don't have a WHDLoad version. If it can be done, how? I am assuming a folder like Games_ADF would be needed?

**Comment by ransom1122** [comment]
What I meant was HoraceTheSpiders whdload games that launch from emulation station via amiberry. Can we also launch adf games as well?

**Comment by a7mag3ddon** [comment]
They already do.

**Comment by osvaldolove** [comment]
Hi - this is more a question for the UAEConfigMaker project, it's not a
part of Amiberry :)
Try logging your question here:
https://github.com/HoraceAndTheSpider/UAEConfigMaker

**Comment by HoraceAndTheSpider** [comment]
Adf's are supported in the RetroPie Amiberry install through a "base config".

The UAE config maker does not support generating .uae files for ADFs, no.

**Comment by ransom1122** [comment]
> The UAE config maker does not support generating .uae files for ADFs, no.

So then is it possible to manually add a new config for an adf game, so we can launch that adf from emulation station? 

I am not sure what you mean by "base config" Quickstart A500 and insert adf in df0: ?

**Comment by blinkydoos** [comment]
If you're using the latest Retropie then selecting a ADF or ZIP from emulationstation should automatically start it. It will even detect mutilple adf's inside zip files (if they have Disk1, Disk2 or something similar in the filename) and insert them. If the file has AGA in the filename then it will run it as an A1200 configuration. If you open the GUI you will see the two RetroPie config's, one for A500 and A1200.

**Comment by ransom1122** [comment]
What folders do the adfs or zips go that emulation station can detect?

Can we also download covers for these adf games like horaces whdload_games covers from openretro?

**Comment by blinkydoos** [comment]
They should be in roms/amiga (the same way all roms should be placed 'roms/[system name]'). I have no idea about scraping though.

**Comment by HoraceAndTheSpider** [comment]
No, my Amiga cover scraper only looks for .uae files in roms/amiga , it doesn't scrape ADFs because the Common TOSEC naming converntion is horrible to work with.

**Comment by ransom1122** [comment]
> No, my Amiga cover scraper only looks for .uae files in roms/amiga , it doesn't scrape ADFs because the Common TOSEC naming converntion is horrible to work with.

Would it be possible to somehow manually add covers to the games with missing covers or adf games?

**Comment by HoraceAndTheSpider** [comment]
I have answered about missing games covers on the UAGS GitHub already?

To "manually" grab covers you need to add an entry to the XML and the artwork folder.

**Comment by ransom1122** [comment]
I looked at the **gamelist.xml** file which was located in **/opt/retropie/configs/all/emulationstation/gamelists/amiga** file size is only 37KB, upon opening it with notepad++ I noticed it only had 511 lines and no where near the full amount of games that are present in Emulation Station?

In regards to adding an ADF, Do i just copy paste for example everything in the red box, and change the details to the ADF game instead? When launching ADF's from emulation station they do not have their own config file as it uses the basic built in a500 config.

So for example adding adf cfg do i change all ADF Game Configs to;;
<path>./Bomb Pac [CD32] [AGA].uae</path>
to
<path>./rp-a500.uae</path>

![1](https://cloud.githubusercontent.com/assets/28079950/26288441/e99a5f10-3ed5-11e7-85fa-a46f6ea569f0.jpg)

**Comment by ransom1122** [comment]
I added Tunnles of Armageddon entry in the gamelist.xml with notepad++ like this;
![1](https://cloud.githubusercontent.com/assets/28079950/26289299/38aae0e6-3ee2-11e7-8e0a-fdfd3036c9d0.jpg)

The path for this game is the a1200 config. I changed it via emulation station to boot from a1200 not a500, as the a500 config causes graphic problems. All the rest looks fine, pictures manually scraped from openretro and put in the **/home/pi/RetroPie/roms/amiga/boxart** folder.

Restart Emulation Station, and the cover and description for the game does not show? like this;
![20170522_113431_resized](https://cloud.githubusercontent.com/assets/28079950/26289379/ee96d252-3ee2-11e7-8856-99bdb3624d54.jpg)

Please can you explain how to add the data manually into the gamelist.xml. and what folder path the gamelist.xml is actually in?
Mine is in /opt/retropie/configs/all/emulationstation/gamelists/amiga/gamelist.xml (but only 38kb) ?
